textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexsplit/RegexSplitOperator.java

```diff
@@ -0,0 +1,185 @@
+            if (bufferCursor == outputTupleBuffer.size()) {
+                hasOutputBufferTuple = false;
+                bufferCursor = 0;
+            }
+            return resultTuple;
+        }
+        return null;
+    }
+    
+    
+    public void populateOutputBuffer(ITuple inputTuple) throws TextDBException {        
+        if (inputTuple != null)
+        {
+            String strToSplit;
+            FieldType fieldType = this.inputSchema.getAttribute(predicate.getAttributeToSplit()).getFieldType();
+            if (fieldType == FieldType.TEXT || fieldType == FieldType.STRING)
+            {
+                strToSplit = inputTuple.getField(predicate.getAttributeToSplit()).getValue().toString();
+                List<String> splitTextList = getSplitText(strToSplit);
+                outputTupleBuffer = new ArrayList<>();
+                for (String splitText : splitTextList) {
+                    List<IField> tupleFieldList = new ArrayList<>();
+                    for (String attributeName : inputSchema.getAttributeNames()) {
+                        if (attributeName.equals(predicate.getAttributeToSplit())) {
+                            if (fieldType == FieldType.TEXT) {
+                                tupleFieldList.add(new TextField(splitText));
+                            } else {
+                                tupleFieldList.add(new StringField(splitText));
+                            }
+                        } else {
+                            tupleFieldList.add(inputTuple.getField(attributeName));
+                        }
+                    }
+                    outputTupleBuffer.add(new DataTuple(inputSchema, tupleFieldList.stream().toArray(IField[]::new)));
+                }
+                hasOutputBufferTuple = true;       //must has a buffer
+                bufferCursor = 0;
+            }
+        }
+    }
+
+    @Override
+    protected void cleanUp() throws TextDBException {
+        hasOutputBufferTuple = false;
+        bufferCursor = 0;
+    }
+    
+    public RegexSplitPredicate getPredicate() {
+        return this.predicate;
+    }
+    
+    /*
+     *  Get a Tuple list from input file operator.
+     */
+    public List<String> getSplitText(String strText) throws TextDBException {
+        List<String> splitTextList = new ArrayList<>();
+        //Create a pattern using regex.
+        Pattern p = Pattern.compile(predicate.getRegex());
+        
+        // Match the pattern in the text.
+        Matcher startM = p.matcher(strText);
+
+        List<Integer> splitIndex = new ArrayList<Integer>();
+        splitIndex.add(0);
+
+        while(startM.find()){
+            if (predicate.getSplitType() == RegexSplitPredicate.SplitType.GROUP_RIGHT) {
+                splitIndex.add(startM.start());
+            } else if (predicate.getSplitType() == RegexSplitPredicate.SplitType.GROUP_LEFT) {
+                splitIndex.add(startM.end());
```

Comment:
yes. if (splitIndex.get(i) != splitIndex.get(i+1)) {
                splitTextList.add(strText.substring(splitIndex.get(i), splitIndex.get(i + 1)));
            },    this will check if the indexs are repeated.